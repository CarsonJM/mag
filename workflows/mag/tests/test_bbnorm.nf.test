nextflow_workflow {

    name "Test workflow: MAG"
    script "../main.nf"
    workflow "MAG"
    tag "workflows"
    tag "mag"
    tag "mag_test_bbnorm"

    test("Parameters: bbnorm = true") {

        when {
            params {
                input                       = 'https://raw.githubusercontent.com/nf-core/test-datasets/mag/samplesheets/samplesheet.csv'
                bbnorm                      = true
                keep_phix                   = true
                skip_adapter_trimming       = true
                centrifuge_db               = null
                kraken2_db                  = null
                skip_krona                  = true
                skip_megahit                = true
                skip_spades                 = true
                skip_spadeshybrid           = true
                skip_quast                  = true
                skip_prodigal               = true
                skip_binning                = true
                skip_binqc                  = true
                skip_gtdbtk                 = true
                skip_prokka                 = true
                skip_metaeuk                = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out,
                ).match() }
            )
        }
    }
}
