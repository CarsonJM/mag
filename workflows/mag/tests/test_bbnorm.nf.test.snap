{
    "Parameters: bbnorm = true": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "test_minigut_run0_fastp_1.fastp.fastq.gz:md5,bac85d89ef44638200c247bc941a66fb",
                            "test_minigut_run0_fastp_2.fastp.fastq.gz:md5,be6b3eb0d336c333e97d925b0e3a0ff6"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "test_minigut_sample2_run0_fastp_1.fastp.fastq.gz:md5,25580ab85c8288a6a7a0b2100d93f1d0",
                            "test_minigut_sample2_run0_fastp_2.fastp.fastq.gz:md5,a91bde6d5c85dd6b47f5bc5cfab65572"
                        ]
                    ]
                ],
                "1": [
                    
                ],
                "10": [
                    "versions.yml:md5,15111ae1ae8b8b570a2efddd7c747bd0",
                    "versions.yml:md5,67cbab915f2801633ea61219086a7735",
                    "versions.yml:md5,67cbab915f2801633ea61219086a7735",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0"
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    
                ],
                "assemblies": [
                    
                ],
                "bin_summary": [
                    
                ],
                "cat": [
                    
                ],
                "genomad": [
                    
                ],
                "metaeuk": [
                    
                ],
                "prodigal": [
                    
                ],
                "prokka": [
                    
                ],
                "refined_bins": [
                    
                ],
                "refined_unbins": [
                    
                ],
                "short_reads": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "test_minigut_run0_fastp_1.fastp.fastq.gz:md5,bac85d89ef44638200c247bc941a66fb",
                            "test_minigut_run0_fastp_2.fastp.fastq.gz:md5,be6b3eb0d336c333e97d925b0e3a0ff6"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "test_minigut_sample2_run0_fastp_1.fastp.fastq.gz:md5,25580ab85c8288a6a7a0b2100d93f1d0",
                            "test_minigut_sample2_run0_fastp_2.fastp.fastq.gz:md5,a91bde6d5c85dd6b47f5bc5cfab65572"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15111ae1ae8b8b570a2efddd7c747bd0",
                    "versions.yml:md5,67cbab915f2801633ea61219086a7735",
                    "versions.yml:md5,67cbab915f2801633ea61219086a7735",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0"
                ]
            }
        ],
        "timestamp": "2024-01-17T09:19:27.939943074"
    }
}